### PR TITLE
[DOCS] Last part of value types.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -398,9 +398,27 @@ a non-rational number).
 String Literals
 ---------------
 
-String literals are written with either double or single-quotes (``"foo"`` or ``'bar'``).  They do not imply trailing zeroes as in C; ``"foo"`` represents three bytes not four.  As with integer literals, their type can vary, but they are implicitly convertible to ``bytes1``, ..., ``bytes32``, if they fit, to ``bytes`` and to ``string``.
+String literals are written with either double or single-quotes (``"foo"`` or ``'bar'``).  They do not imply trailing zeroes as in C; ``"foo"`` represents three bytes (not four).  As with integer literals, their type can vary, but they are implicitly convertible to ``bytes1``, ..., ``bytes32``, if they fit, to ``bytes`` and to ``string``.
```
Comment (axic) [inline docs/types.rst]: Why not `represents three bytes, not four`?

File: docs/types.rst [hunk 2]
```diff
@@ -398,9 +398,27 @@ a non-rational number).
...
+ - ``\n`` (newline)
+ - ``\r`` (carriage return)
+ - ``\t`` (tab)
+ - ``\v`` (vertical tab)
+ - ``\xNN`` (hex escape, see below)
+ - ``\uNNNN`` (unicode escape, see below)
+
+ ``\xNN`` takes a hex value and inserts the appropriate byte, while ``\uNNNN`` takes a Unicode codepoint and inserts an UTF-8 sequence.
+
+Any unicode line terminator which is not a newline (i.e. LF, VF, FF, CR, NEL, LS, PS) is considered to
```
Comment (axic) [inline docs/types.rst]: IIRC there is another part of the documentation mentioning this.

File: docs/types.rst [hunk 3]
```diff
@@ -398,9 +398,27 @@ a non-rational number).
 String Literals
 ---------------
 
-String literals are written with either double or single-quotes (``"foo"`` or ``'bar'``).  They do not imply trailing zeroes as in C; ``"foo"`` represents three bytes not four.  As with integer literals, their type can vary, but they are implicitly convertible to ``bytes1``, ..., ``bytes32``, if they fit, to ``bytes`` and to ``string``.
+String literals are written with either double or single-quotes (``"foo"`` or ``'bar'``).  They do not imply trailing zeroes as in C; ``"foo"`` represents three bytes (not four).  As with integer literals, their type can vary, but they are implicitly convertible to ``bytes1``, ..., ``bytes32``, if they fit, to ``bytes`` and to ``string``.
 
-String literals support escape characters, such as ``\n``, ``\xNN`` and ``\uNNNN``. ``\xNN`` takes a hex value and inserts the appropriate byte, while ``\uNNNN`` takes a Unicode codepoint and inserts an UTF-8 sequence.
+String literals support the following escape characters:
+
+ - ``\<newline>`` (escapes an actual newline)
```
Comment (axic) [inline docs/types.rst]: I think we should add an example for this.

File: docs/types.rst [hunk 4]
```diff
@@ -488,6 +507,11 @@ omitted. Note that this only applies to function types. Visibility has
 to be specified explicitly for functions defined in contracts, they
 do not have a default.
 
+Conversions:
+
+A value of external function type can be explicitly converted to ``address``
+resulting in the address of the contract of the function.
```
Comment (axic) [inline docs/types.rst:523]: Hm, can function types be payable and if so do they cast to `address payable`?

File: docs/types.rst (hunk 3 above)
Comment (chriseth) [reply docs/types.rst]: Added example.

File: docs/types.rst (hunk 4 above)
Comment (chriseth) [reply docs/types.rst:523]: They should always cast to `address`. The fact that one function in the contract is payable does not mean that the contract will have a payable fallback function.

Comment (chriseth) [reply docs/types.rst:523]: I'll check if we have a test.

Comment (chriseth) [reply docs/types.rst:523]: It seems this was a bug. Created PR: https://github.com/ethereum/solidity/pull/5030

File: docs/types.rst [hunk 5]
```diff
@@ -398,9 +398,37 @@ a non-rational number).
...
+ - ``\b`` (backspace)
+ - ``\f`` (form feed)
+ - ``\n`` (newline)
+ - ``\r`` (carriage return)
+ - ``\t`` (tab)
+ - ``\v`` (vertical tab)
+ - ``\xNN`` (hex escape, see below)
+ - ``\uNNNN`` (unicode escape, see below)
+
+``\xNN`` takes a hex value and inserts the appropriate byte, while ``\uNNNN`` takes a Unicode codepoint and inserts an UTF-8 sequence.
```
Comment (erak) [inline docs/types.rst:418]: Should we add a note about ``0X`` being disallowed from 0.5.0 on?

Comment (chriseth) [reply docs/types.rst:418]: We should, but this is not the place. `0X` was used for number / bytes constants. This is a hex escape inside a string literal.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**